### ISSUE TITLE
Action binding doesn't refresh with ApplyBindingOverride() during runtime

### ISSUE THREAD

**Issue by borishsu** [Unity-Technologies/InputSystem#247]
I want a way to let users select a different button to trigger the fire action. Take the code example below, when I press `num 1` then the original `A` key will be replaced with `W` key for that action. 

```csharp
public class TestBindingOverride: MonoBehaviour
{
    public InputAction fireAction;

    public void Awake()
    {
        fireAction = new InputAction(name: "fire", binding: "<Keyboard>/a");

        fireAction.performed +=
            ctx =>
            {
                Debug.Log("fire!");
            };   
    } 

    public void OnEnable()
    {
        fireAction.Enable();
    }

    public void OnDisable()
    {
        fireAction.Disable();
    }

    public void Update()
    {
        if (Input.GetKeyDown(KeyCode.Alpha1))
        {
            fireAction.Disable();
            fireAction.ApplyBindingOverride(path: "<Keyboard>/a", newBinding: "<Keyboard>/w");
            fireAction.Enable();
        }
    }
}
```

The problem is that the binding override doesn't apply immediately, and I found it refreshed when any device is discovered and added to the system.

This is when I press `num 1` to call ApplyBindingOverride(), but nothing happened. The `A` key still triggers the fire action but `W` key doesn't.
![2018-09-12 14_18_19](https://user-images.githubusercontent.com/38482575/45407428-fa47dd80-b69b-11e8-972f-663e79f59832.png)

When I turn on my Vive tracker (any other device will do), the system updates the binding from `A` key to `W` key. Now `W` key triggers the fire action but `A` key doesn't.
![2018-09-12 14_18_46](https://user-images.githubusercontent.com/38482575/45407548-4abf3b00-b69c-11e8-912b-532f3916c8e0.png)

Any idea on how to fix it? Or is there any better way to do this? Thanks.

**Comment by Rene-Damm** [comment]
Confirmed. `InputActionRebindingExtensions` isn't flushing resolved data.

Partial fix in 07b5979. Needs some related changes to happen to catch all the cases.

**Comment by JiaUnity** [comment]
Verified fix on 2019.1.0b9 with Develop (16f1ab7)
